File: 인터넷 강의/python_ruby/Override/1.py
class C1:
    def m(self):
        return "parent"


class C2(C1):
    # Override
    def m(self):
        # * C1의 m을 가르킨다. 
        # super().m()
        return super().m() + " child"
    # 메소드가 존재하지 않는 클래스는
    # ! "pass"키워드 사용
    pass

o = C2()
print(o.m())

File: 인터넷 강의/python_ruby/Override/1.rb
class C1
    def m()
        return "parent"
    end
end

class C2 < C1
    def m()
        # ! ruby에서 super는 현재 메소드에서의 같은 이름을 갖고있는 상위 클래스의 "메소드"를 가르킨다!
        return super() + " child"
    end
end

o = C2.new()
p o.m()